ray-on-gke/tpu/kuberay-tpu-webhook/webhook_main_test.go

```diff
@@ -357,142 +413,44 @@ func Test_GetReplicaIndex(t *testing.T) {
 			additionalNumOfHosts:  2,
 			additionalNumReplicas: 3,
 		},
-		"deleted pods from replica": {
-			// should re-assign pods to lowest index replicas with # isCreated pods < NumOfHosts
-			numOfHosts:      4,
-			numReplicas:     4,
-			workersToDelete: []worker{worker{0, 0, true}, worker{2, 1, true}, worker{3, 2, true}},
-		},
-		"delete pods from different multi-host groups": {
-			// pods should be reassigned the lowest replica ID with # isCreated pods < NumOfHosts
-			// in each respective worker group
-			numOfHosts:            4,
-			numReplicas:           4,
-			additionalGroupStr:    "another-worker-group",
-			additionalNumOfHosts:  4,
-			additionalNumReplicas: 3,
-			workersToDelete:       []worker{worker{1, 0, true}, worker{2, 1, true}, worker{3, 2, true}},
-		},
 	}
 
 	// validate getReplicaIndex() returns the expected Replica ID for TPU pods in varying pod slices
 	for name, tc := range tests {
 		t.Run(name, func(t *testing.T) {
-			sliceToWorkersCopy := deepCopySliceToWorkers()
+			sliceToWorkerIDs := make(map[slice][]int)
 			for i := 0; i < tc.numReplicas; i++ {
 				testPodSlice := slice{instanceName, groupNameStr, namespaceStr, i, tc.numOfHosts}
 				for j := 0; j < int(tc.numOfHosts); j++ {
-					replicaIndex := getReplicaIndex(instanceName, groupNameStr, namespaceStr)
+					replicaIndex := getReplicaIndex(sliceToWorkerIDs, instanceName, groupNameStr, namespaceStr)
```

Comment:
I think we could simplify this test by just having the existing `sliceToWorkerIDs` pre-populated by the test and just asserting the returned next index?

Reply:
```go
t.Run(tc.name, func(t *testing.T) {
    replicaIndex := getReplicaIndex(tc.sliceToWorkerIDs, instanceName, groupNameStr, namespaceStr)
    if replicaIndex != tc.expectedReplicaIndex {
       t.Error("unexpected next replica index")
    }
})
```

Reply:
Done in [b70535d](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/723/commits/b70535d96c495b0afafbfe07cd8bfa82a3d0d5b8), I refactored `Test_GetNextWorkerID` to follow the same pattern.